packages/build-config/plugins/webpack-service-worker.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var SingleEntryPlugin = require('webpack/lib/SingleEntryPlugin');
+var RawSource = require('webpack-sources').RawSource;
+
+var hopsConfig = require('hops-config');
+
+var PLUGIN_NAME = 'hops-service-worker-plugin';
+
+module.exports = function ServiceWorkerPlugin() {
+  var assetPath = hopsConfig.workerPath.replace(/^\/+/, '');
+
+  this.apply = function(compiler) {
+    if (!hopsConfig.workerFile) {
+      return;
+    }
+
+    function onMake(compilation, callback) {
+      compilation
+        .createChildCompiler(PLUGIN_NAME, { filename: assetPath }, [
+          new SingleEntryPlugin(
+            compiler.context,
+            require.resolve('../shims/worker-shim'),
+            'worker'
+          ),
+        ])
+        .runAsChild(callback);
+    }
+
+    function onEmit(compilation, callback) {
+      compilation.assets[assetPath] = new RawSource(
```

Comment:
Maybe this should also contain `locations`? Or we fetch `locations` inside Service Worker directly by importing `hopsConfig.locations`.

Reply:
+1 for hopsConfig. It's how you'd access locations anywhere else too, so I guess it make sense to have it consistent.